baseplate/server/__init__.py

```diff
@@ -265,6 +265,13 @@ def load_app_and_run_server() -> None:
         config = read_config(args.config_file, args.server_name, args.app_name)
     assert config.server
 
+    if is_metrics_enabled(config.app):
```

Comment:
What's the reasoning behind moving this logic?

Reply:
We generate metrics even if `is_metrics_enabled` is enabled with the previous logic. So without this move if you `ls` the `prometheus multiprocess dir` you see that there are files generated there because prom is initialized

Reply:
Actually let me answer a bit better:

Somewhere in the app construction we are generating data prometheus stats before  we call `is_metrics_enabled` and `start_prometheus_exporter()`. This can be verified by the presence of files in the prometheus dir.

And because we set  multiprocess client id at `start_prometheus_exporter` these metrics are using the default `id` which defeats the purpose of this PR

Reply:
Thanks for explaining! A comment to that effect might be good so we don't make that mistake again :)